nestjs/src/courses/mentors/dto/mentor-options.dto.ts

```diff
@@ -0,0 +1,50 @@
+import { Student } from '@entities/index';
+import { Mentor } from '@entities/mentor';
+import { PreferredStudentsLocation } from '@common/enums/mentor';
+import { ApiProperty, ApiResponse } from '@nestjs/swagger';
+import { IsArray, IsNumber, IsString } from 'class-validator';
+import { PersonDto } from 'src/core/dto';
+
+class StudentsDto {
+  @ApiProperty()
+  @IsNumber()
+  id: number;
+
+  @ApiProperty()
+  @IsString()
+  githubId: string;
+
+  @ApiProperty()
+  @IsString()
+  name: string;
+
+  constructor(student: Student & { user: { githubId: string; firstName: string; lastName: string } }) {
+    this.id = student.id;
+    this.githubId = student.user.githubId;
+    this.name = PersonDto.getName(student.user);
+  }
+}
+
+@ApiResponse({})
+export class MentorOptionsDto {
+  constructor(
+    mentor: Omit<Mentor, 'students'> & {
```

Comment:
You could use extracted student type here, also could extract mentor type to use it in mentor service `getMentorOptions`

Reply:
thx, done